src/worker/connectionincoming.js

```diff
@@ -279,4 +311,29 @@ class ConnectionIncoming {
     }
 }
 
+// Split []tokens into batches that when joined with prefix and suffix, its length does not exceed maxLen
+// eg. ['one', 'two', ...100 more tokens] = [['one', 'two'], [...more tokens]]
+function batchIsupportTokensToMaxLenth(tokens, prefix, suffix='is supported by this server', maxLen=512) {
+    // 1 = the extra space after the prefix
+    let l = prefix.length + 1;
+    // 1 = the : before the suffix
+    l += suffix.length + 1; 
```

Comment:
Should we add an additional 2 for `\r\n`? I noticed below we have a `+1` but not sure whether that's for the space between the `005` and params or for just the `\n` alone.